scripts/o3de/o3de/engine_template.py

```diff
@@ -1511,21 +1511,18 @@ def create_project(project_path: pathlib.Path,
 
     # project restricted name
     if project_restricted_name and not project_restricted_path:
-        project_restricted_path = manifest.get_registered(restricted_name=project_restricted_name)
-
+        gem_restricted_path = manifest.get_registered(restricted_name=project_restricted_name)
+        if not gem_restricted_path:
+            logger.error(f'Project Restricted Name {project_restricted_name} cannot be found.')
+            return 1
     # project restricted path
     elif project_restricted_path:
         if not os.path.isabs(project_restricted_path):
-            default_projects_restricted_folder = manifest.get_registered(restricted_name='projects')
-            new_project_restricted_path = default_projects_restricted_folder/ project_restricted_path
-            logger.info(f'Project restricted path {project_restricted_path} is not a full path, we must assume its'
-                        f' relative to default projects restricted path = {new_project_restricted_path}')
-            project_restricted_path = new_project_restricted_path
-    elif template_restricted_path:
-        project_restricted_default_path = manifest.get_registered(restricted_name='projects')
-        logger.info(f'--project-restricted-path is not specified, using default project restricted path / project name'
-                    f' = {project_restricted_default_path}')
-        project_restricted_path = project_restricted_default_path
+            logger.error(f'Project Restricted Path {project_restricted_path} is not an absolute path.')
+            return 1
+    # neither make a dir side by side with the new project named <project_name>-restricted
+    else:
+        project_restricted_path = project_path.parent / (project_name + '-restricted')
```

Comment:
Is there any file name validation for 'project_name' (like having invalid path name characters like >, < on windows, etc). 

Also, suggest to use f-strings for the filename as well
```suggestion
        project_restricted_path = project_path.parent / f'{project_name}-restricted'
```

Reply:
Yes the project-name is validated at line [1503](https://github.com/o3de/o3de/pull/4337/files/25307f78a7cfa42fe0730765827bc29d44ff13ca#diff-5cfe7434ea975d8c4a43d56e98dd1acfa63d3b6f2a29a6c65ccd3704eb243395R1503)

Reply:
[1503](https://github.com/o3de/o3de/pull/4337/files/25307f78a7cfa42fe0730765827bc29d44ff13ca#diff-5cfe7434ea975d8c4a43d56e98dd1acfa63d3b6f2a29a6c65ccd3704eb243395R1503), just out of reach of this diff window

Reply:
done